### PR TITLE
adding an audio_sample_url method and ISBN13 

### PR DESCRIPTION
**Pull request by hosseintoussi**

### CODE REVIEW

File: lib/onix/collateral_detail.rb [hunk 1]
```diff
@@ -72,6 +72,13 @@ def epub_sample_resource
       end
     end
 
+    def audio_sample_url
+      audio = @supporting_resources.audio.first.versions.last
+      if audio
```
Comment (zfhui) [inline lib/onix/collateral_detail.rb]: You could do `if audio = @supporting_resources.audio.first.versions.last` directly. 

The right side of the `=` looks like it's breaking the law of Demeter, maybe it makes sens to implement a method on `@supporting_resources.audio` which takes care of the rest?

File: lib/onix/onix21.rb [hunk 2]
```diff
@@ -559,14 +560,38 @@ def distributor_name
...
       def frontcover_url
-        fc=@media_files.select { |mf| mf.media_file_type_code.human=="ImageFrontCover" && mf.media_file_link_type_code.human=="Url"}
+        fc=@media_files.select do |mf|
+          mf.media_file_type_code.human=="ImageFrontCover" && mf.media_file_link_type_code.human=="Url"
+        end
         if fc.length > 0
           fc.first.link
         else
           nil
         end
```
Comment (zfhui) [inline lib/onix/onix21.rb]: Not your fault, but suggestion to refactor this line into: 
```
return unless fc.length>0

fc.link.link
```

File: lib/onix/onix21.rb [hunk 3]
```diff
@@ -559,14 +560,38 @@ def distributor_name
       end
 
       def frontcover_url
-        fc=@media_files.select { |mf| mf.media_file_type_code.human=="ImageFrontCover" && mf.media_file_link_type_code.human=="Url"}
+        fc=@media_files.select do |mf|
+          mf.media_file_type_code.human=="ImageFrontCover" && mf.media_file_link_type_code.human=="Url"
+        end
```
Comment (zfhui) [inline lib/onix/onix21.rb]: It looks like this line could be a method on `@media_files` or a private method here with verbose name. What do you think?

File: lib/onix/onix21.rb [hunk 4]
```diff
@@ -559,14 +560,38 @@ def distributor_name
...
+      def high_quality_frontcover_url
+        fc=@media_files.select do |mf|
+          mf.media_file_type_code.human=="ImageFrontCoverHighQuality" && mf.media_file_link_type_code.human=="Url"
+        end
+        if fc.length > 0
+          fc.first.link
+        else
+          nil
+        end
+      end
```
Comment (zfhui) [inline lib/onix/onix21.rb]: Same thing here, I think, we can make this prettier! ✨

File: lib/onix/onix21.rb [hunk 5]
```diff
@@ -559,14 +560,38 @@ def distributor_name
...
+      def audio_sample_url
+        audio=@media_files.select do |mf|
+          mf.media_file_type_code.human=="AudioSampleContent" && mf.media_file_link_type_code.human=="Url"
+        end
+        if audio.length > 0
+          audio.first.link
+        else
+          nil
+        end
+      end
```
Comment (zfhui) [inline lib/onix/onix21.rb]: Same thing here, I think, we can make this prettier! ✨

File: lib/onix/product.rb [hunk 6]
```diff
@@ -358,6 +358,12 @@ def frontcover_url
       end
     end
 
+    def audio_sample_url
+      if @collateral_detail
+        @collateral_detail.audio_sample_url
+      end
```
Comment (zfhui) [inline lib/onix/product.rb]: Suggestion to use `@collateral_detail&.audio_sample_url` instead.

**Comment by hosseintoussi** [comment]
@zfhui, I agree with you on all the points here. But I'm trying to keep the same style of code here as I think writing new codes with a different style will lead to more confusion and it will become harder to maintain in near future. If it's OK with you i'd like to keep everything the same way.

**Pull request merged**